maestro-agent/src/main/java/org/maestro/agent/base/MaestroAgent.java

```diff
@@ -18,98 +21,140 @@
     }
 
     public void handle(StopReceiver note) {
-
+        File entryPointDir = new File(path, "stopreceiver");
+        callbacksWrapper(entryPointDir);
     }
 
     public void handle(StopSender note) {
-
+        File entryPointDir = new File(path, "stopsender");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(StatsRequest note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "stats");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(FlushRequest note) {
         super.handle(note);
 
-        URL uri = this.getClass().getResource("/org/maestro/agent/ext/requests/flush/");
-        if (uri != null) {
-            String path = uri.getPath();
-            File file = new File(path);
-            try {
-                groovyHandler.setInitialPath(file);
-                groovyHandler.runCallbacks();
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        }
-        else{
-            logger.error("Unable to load files for extension points.");
-        }
-
-
+        File entryPointDir = new File(path, "flush");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(Halt note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "halt");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(SetRequest note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "set");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(TestFailedNotification note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "notifyfail");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(TestSuccessfulNotification note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "notifysuccess");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(AbnormalDisconnect note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "abnormaldisconnect");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(PingRequest note) throws MaestroConnectionException, MalformedNoteException {
         super.handle(note);
 
+        File entryPointDir = new File(path, "pingrequest");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(GetRequest note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "get");
+        callbacksWrapper(entryPointDir);
+    }
+
+    private void callbacksWrapper(File entryPointDir) {
+        try {
+            groovyHandler.setInitialPath(entryPointDir);
+            groovyHandler.runCallbacks();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
@Frawless I was thinking about this part of the code. I think we need 2 things here: 1 to log what happened and 2 the agent needs to send an internal error notification.